docs/en/stores/basic-usage.md

```diff
@@ -0,0 +1,471 @@
+
+> main.ts
+
+```ts
+import Registry from '@dojo/framework/widget-core/Registry';
+import Store from '@dojo/framework/stores/Store';
+import { State } from './interfaces';
+
+const registry = new Registry();
+const store = new Store<State>();
+registry.defineInjector('state', (invalidator) => {
+	store.on('invalidate', () => invalidator());
+	return () => store;
+});
+```
+
+`State` defines the shape of the global store using an interface. In general state should be serializable. This improves performance and makes it easier for Dojo to determine when changes to the data occur.
+
+> interfaces.d.ts
+
+```ts
+interface User {
+	id: string;
+	name: string;
+}
+
+export interface State {
+	auth: {
+		token: string;
+	};
+	users: {
+		current: User;
+		list: User[];
+	};
+}
+```
+
+The above is a simple example for illustrative purposes. Examples in this reference guide will refer back to this interface to reduce repetition.
+
+## Updating stores
+
+To work with Dojo Stores there are three core concepts.
+
+-   **Operations** are instructions to manipulate the state held by the store
+-   **Commands** are simple functions that perform business logic and return operations
+-   **Processes** execute a group of commands and represent application behavior
+
+### Operations
+
+Stores cannot be modified directly. Instead four operations are available for modifying state: `add`, `remove`, `replace`, and `test`. These operations are represented by simple objects based on the [JSON patch](http://jsonpatch.com/) format, but in practice helper functions will always be used.
+
+#### `path` and `at`
+
+The path is a `string` that describes the location where an operation is applied. The `path` and `at` functions are part of the `Store` and are accessible inside of a `Command` (which we will talk about later).
+
+Here are a couple examples to show how `path` and `at` describe a location in the store. The `State` is the same as defined above in `interface.d.ts`. It is used by the `Store` to understand the shape of the state data.
+
+> `path` of the current user name
+
+```ts
+const store = new Store<State>();
+const { path } = store;
+
+path('users', 'current', 'name');
+```
+
+This path refers to the `string` value located at `/users/current/name`. `path` is used to transverse the hierarchy in a type-safe way. It ensures only the property names defined in the `State` interface are used.
+
+> `at` is an offset
+
+```ts
+const store = new Store<State>();
+const { at, path } = store;
+
+at(path('users', 'list'), 1);
+```
+
+This path refers to the `User` located at `/users/list` at offset `1`.
+
+#### `add` operation
+
+Adds a value to an object or inserts it into an array.
+
+> `add` example
+
+```ts
+import { add } from '@dojo/framework/stores/state/operations';
+
+const store = new Store<State>();
+const { apply, at, path } = store;
+const user = { id: '0', name: 'Paul' };
+
+apply([add(at(path('users', 'list'), 0), user)]);
```

Comment:
I don't think we should be demonstrating interacting with the store directly like this. It's not an encouraged pattern